app/src/main/java/ch/epfl/sdp/blindwar/ui/StatisticsActivity.kt

```diff
@@ -1,28 +1,77 @@
 package ch.epfl.sdp.blindwar.ui
 
+import android.content.ContentValues
 import android.os.Bundle
+import android.util.Log
 import android.view.View
-import android.widget.AdapterView
-import android.widget.ArrayAdapter
-import android.widget.Spinner
-import android.widget.Toast
+import android.widget.*
 import androidx.appcompat.app.AppCompatActivity
 import ch.epfl.sdp.blindwar.R
+import ch.epfl.sdp.blindwar.user.AppStatistics
+import ch.epfl.sdp.blindwar.user.Mode
+import ch.epfl.sdp.blindwar.user.Result
+import ch.epfl.sdp.blindwar.user.User
+import com.google.firebase.database.*
+import com.google.firebase.database.ktx.getValue
 
 
 class StatisticsActivity : AppCompatActivity() {
 
-    //Float values for stats (only for demo)
-    private var wins = 45
-    private var losses = 35
-    private var draws = 100 - wins - losses
+    val firebaseDatabase: FirebaseDatabase = FirebaseDatabase.getInstance()
+    val databaseReference = firebaseDatabase.getReference("Data")
+
+    var userStatistics: AppStatistics = AppStatistics()
 
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_statistics)
+
+        val eloView = findViewById<TextView>(R.id.eloExampleView)
+        eloView.text = "eloView"
+        val winView = findViewById<TextView>(R.id.winNumberView)
+        winView.text = "winView"
+        val drawView = findViewById<TextView>(R.id.drawNumberView)
+        drawView.text = "drawView"
+        val lossView = findViewById<TextView>(R.id.lossNumberView)
+        lossView.text = "lossView"
+        val winPercent = findViewById<TextView>(R.id.winPercentView)
+        winPercent.text = "winPercent"
+        val drawPercent = findViewById<TextView>(R.id.drawPercentView)
+        drawPercent.text = "drawPercent"
+        val lossPercent = findViewById<TextView>(R.id.lossPercentView)
+        lossPercent.text = "lossPercent"
+        val correctView = findViewById<TextView>(R.id.correctNumberView)
+        correctView.text = "correctView"
+        val wrongView = findViewById<TextView>(R.id.wrongNumberView)
+        wrongView.text = "wrongView"
+        val correctPercent = findViewById<TextView>(R.id.correctnessPercentView)
+        correctPercent.text = "correctPercent"
+        val wrongPercent = findViewById<TextView>(R.id.wrongPercentView)
+        wrongPercent.text = "wrongPercent"
+
+        databaseReference.addValueEventListener(object : ValueEventListener {
```

Comment:
You can probably use the "addUserListener" function I created for this purpose in UserDatabase.